File: 1-easy/problems/1013-partition-array-into-three-parts-with-equal-sum/solution.ts
function canThreePartsEqualSum(A: any): boolean | number | string | any {
  """
  :type A: List[int]
  :rtype: bool
  """
  total = sum(A)
  if (total % 3 != 0) {
      return false;
  parts, curr = 0, 0
  for (x in A) {
      curr += x
      if (curr == total//3) {
          parts += 1
          curr = 0
  return parts >= 3;
}

export { canThreePartsEqualSum };

File: 1-easy/problems/1013-partition-array-into-three-parts-with-equal-sum/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def canThreePartsEqualSum(self, A):
        """
        :type A: List[int]
        :rtype: bool
        """
        total = sum(A)
        if total % 3 != 0:
            return False
        parts, curr = 0, 0
        for x in A:
            curr += x
            if curr == total//3:
                parts += 1
                curr = 0
        return parts >= 3
